CONRTIBUTING.md

```diff
@@ -0,0 +1,84 @@
+introduced by placing these patterns in a semi-official package:
+
+  - Because a pattern is being considered for inclusion in a package that is
+  intended to be used across many projects, a more rigorous review process
+  seems like a good idea.
+  - Identifying a pattern as one we have already implemented most often occurs
+  during a mad push to meet a deadline, so the person raising the PR requesting
+  the pattern be included in the `simpl` canon needs it merged quickly.
+
+__The following approach aims to reconcile these two competing demands:__
+
+  - The `simpl` package contains two types of modules:
+    - __Core__ modules are well-established, curated patterns that have been
+    thoroughly vetted. They have earned their place at the root level of the
+    `simpl` package. `config`, `exceptions`, `log` are all
+    examples of __Core__ modules.
+    - The `simpl.incubator` module contains modules that are being considered
+    for inclusion in __Core__, but are still being vetted. This solves the
+    copy/paste duplication issue while still allowing time for the rigor we
+    wish to apply to modules before being accepted into __Core__.
+  - Any submitted PR is a candidate for inclusion in __Core__.
+  - If a given PR is being contested, the submitter may opt to move the code
+  into `simpl.incubator` in an effort to keep the vetting process from blocking
+  usage of the code/pattern in other projects.
+  - `simpl.incubator` code can eventually be promoted to __Core__. If this
+  occurs the module will be moved to the root level of the `simpl` directory
+  and will no longer be available in `simpl.incubator`, requiring references to
+  the module to be updated when upgrading to the version of `simpl` that has
+  the module in __Core__.
+
+--------
+
+
+### Dealing with optional dependencies
+
+Some of `simpl`'s modules deal with specific tools (like MongoDB or Redis) and
+leverage Python packages built for those tools. This presents a problem,
+though: what if a project isn't using that particular `simpl` module? We don't
+want an unused module to force a project to install a package it has no
+intention of using!
+
+The following guidelines aim to solve this issue:
+  - When possible, limit package use to the
+  [Python Standard Library](https://docs.python.org/3/library/)
+  - When other packages are absolutely necessary make sure that the dependency
+  remains entirely encapsulated in the module that needs it. This means that
+  even though `simpl.bottle` requires the `bottle` package, neither a warning
+  or exception should be thrown when importing `simpl.log` or `simpl.git` as
+  they can be used just fine without `bottle`.
+  - When other packages are optional (e.g. `simpl.middleware.cors` will use
+  `webob` but it isn't absolutely required), make sure that the module properly
+  handles both cases
+  ([cors.py](https://github.com/checkmate/simpl/blob/master/simpl/middleware/cors.py)
+  provides a good example of this).
```

Comment:
The example we gave is that you could `import simpl.cors` and then either read `CORSMiddleware.default_headers` or subclass it and override the `__call__` method. Not the most visibly useful of examples, but somewhat illustrative of the concept.

Is there a better example already in simpl that you have in mind?